docs/dkt.md

```diff
@@ -0,0 +1,341 @@
+events. Events carry different attributes, depending on the kind of
+the source they come from. For instance events coming from Github
+sources will have a `COMMIT` attributes and might have `PR` related
+attributes as well (such as `PR/number`, `PR/state`, etc.) Note that,
+depending on the attribute, a different hash (sha1 for Git, sha256 for
+Docker) or digest (md5 for paths and hockeyapp) can be used.
+Attributes might be writable (such as `PR/state`).
+
+```
+$ dkt source log
+NAME     HEAD
+pinata   1bf6cb20
+pitfall  b348e952
+cases    e5e433a1
+dmg      0E500A4E
+tests    dedba9be
+```
+
+Depending on their kind, data sources can exist in multiple
+versions. A Git repository on Github will expose all its commits and
+the optional PR it is the head of:
+
+```
+$ dkt source log pinata
+ID       PR COMMIT
+1bf6cb20 -  sha1:1bf6cb2021fd9f38605be50d4ebc9e18b0bf6e90
+3dd2cc1d 42 sha1:3dd2cc1db51828a643237c39d5ed76bb066bcad8
+[..]
+```
+
+Hockeyapp sources contains the list of `dmg` files that the user can
+download with their checksum and build numbers:
+
+```
+$ dkt source log dmg
+ID       BUILD CHECKSUM
+0E500A4E 2246  md5:0E500A4EB2A162F8098575DACED142E1
+82A53D4E 2245  md5:82A53D4EA834FCAC928D2D589E3CF859
+[..]
+```
+
+Data-source can be forced to a specific version (note the `*` sign
+close to the name of the pinned package):
+
+```
+$ dkt source pin pinata 3dd2cc1d
+$ dkt source list
+NAME     HEAD
+pinata*  3dd2cc1d
+pitfall  b348e952
+cases    e5e433a1
+dmg      0E500A4E
+tests    dedba9be
+```
+
+To change the location of a source (to point to a local mirror, for
+instance):
+
+```
+$ dkt source pin pinata /local/git/pinata -k git
+```
+
+> NOTE(samoht): all source by default will subscribe to upstream
+changes to get notified when there are changes. If a data-source is
+locally updated, the change will be automatically pushed to the remote
+source if the source has write access (which is set-up on init, using
+the `--read-write` or the `--write` flag). It is expected that the
+changes are propagated asynchronously so might results in conflicts:
+each data-sources will have a built-in default conflict resolution
+mechanism but the user might be able to overwrite them and manually
+resolve confict. TBD
+
+Out-of-band relations can also be accounted in this system. For
+instance, to relate a particular `dmg` stored in hockeyapp and the
+Git commit it was generated from, use:
+
+```
+dkt source link pinata:3dd2cc1d dmg:0E500A4E
+```
+
+Although ideally, this relation should be managed directly by datakit.
+
+### Compute nodes
+
+Compute nodes are temporary process executing arbitrary
+commands. Their input data sources are put in scope before the command
+are run. The compute nodes are expected to put their results in their
+output data sources: that's the only way they have to persist
+state. Compute node can have OS/architecture constraints, and can be
+successful or fail.
```

Comment:
Do they get access to the source itself, or just the current commit from that source? Same for output.